packer/builder/azure/arm/config.go

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/Godeps/_workspace/src/github.com/Azure/go-autorest/autorest/to"
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/communicator"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+	"github.com/mitchellh/packer/template/interpolate"
+)
+
+const (
+	DefaultUserName = "packer"
+	DefaultVMSize   = "Standard_A1"
+)
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+
+	// Authentication via OAUTH
+	ClientID       string `mapstructure:"client_id"`
+	ClientSecret   string `mapstructure:"client_secret"`
+	TenantID       string `mapstructure:"tenant_id"`
+	SubscriptionID string `mapstructure:"subscription_id"`
+
+	// Capture
+	CaptureNamePrefix    string `mapstructure:"capture_name_prefix"`
+	CaptureContainerName string `mapstructure:"capture_container_name"`
+
+	// Compute
+	ImagePublisher string `mapstructure:"image_publisher"`
+	ImageOffer     string `mapstructure:"image_offer"`
+	ImageSku       string `mapstructure:"image_sku"`
+	Location       string `mapstructure:"location"`
+	VMSize         string `mapstructure:"vm_size"`
+
+	// Deployment
+	ResourceGroupName string `mapstructure:"resource_group_name"`
```

Comment:
There are two resource groups.  The one containing the storage account where the VM's disk will reside, and the temporary one that contains everything else.  This is why we differentiate.

I did not call it StorageAccountResourceGroupName because it sounded long winded, and confusing.  I was going to strategize with you on ways to remove the need for this in the future.